### DISCUSSION TITLE
Lets Encrypt

### DISCUSSION THREAD

**Discussion by fsfc-sysadmin** [Ylianst/MeshCentral#4503] [Q&A]
Hi guys ,

I am wondering how to verify if meshcentral auto renew letsencypt certificate ? I've tried le and lecheck and i noticed that "daysleft" keep on decreasing . How to verify if meshcentral does auto-renew lets encrypt certificate ? 
![image](https://user-images.githubusercontent.com/109936671/188417768-9adc3b6f-c5fc-42d2-8ec7-9bb1404e26e5.png)

![image](https://user-images.githubusercontent.com/109936671/188417341-d52140fc-58ee-462f-84ab-e78e5ee49805.png)

**Comment by si458** [comment]
Meshcentral will automatically renew the certificate if its less than 30 days I believe as that's standard let's Encrypt policy for everyone, 
You will also notice the days left should change bk to 90 when it renews and then start counting down again

**Reply by fsfc-sysadmin** [reply]
Thank you , would it break if we have added a custom port after the certificate was generated ? So instead of 443 , we use different port .

**Reply by fsfc-sysadmin** [reply]
i will try to observer this , soo it should be renewing the certificate when it is below 30 days .

**Reply by nccerhostmaster** [reply]
@jsinoy if you want to use non-standard ports with Lets Encrypt you may want to look at using DNS challenges.

**Comment by fsfc-sysadmin** [comment]
![image](https://user-images.githubusercontent.com/109936671/188423094-7bad0b3f-5e8d-401c-a34e-a644721db85c.png)

**Comment by fsfc-sysadmin** [comment]
Failed to obtain certificate.
![image](https://user-images.githubusercontent.com/109936671/188428178-9dabc47f-224a-4f94-84ce-a95e5a1d0ac7.png)

**Comment by silversword411** [comment]
For LE renewal you need port 80 exposed publicly for the auto-renewals done by certbot. If you're using something other than 443 for your web server, I'm not sure if it'll work or not. 

If you needed LE wildcards, you'll need DNS based LE renewal which is a manual process

**Reply by dinger1986** [reply]
a custom port is fine as long as port 80 is still open

**Comment by Ylianst** [comment]
The MeshCentral redirection port needs to be open to the internet and listening to port 80. So, you need this (which is the default):

```
"redirport": 80
```

**Comment by silversword411** [comment]
Looks like all this info is already in the docs

https://ylianst.github.io/MeshCentral/meshcentral/#lets-encrypt-support

**Comment by fsfc-sysadmin** [comment]
Thanks guys . its working now.

Cheers,
